client/charts/js/components/DataLoader.jsx

```diff
@@ -8,39 +8,87 @@ function Loader() {
 }
 
 function DataLoader({ dataUrl, dataParser, dataKey, loadingComponent, children }) {
-	const [data, setData] = useState([])
+	const [data, setData] = useState({})
 	const [loading, setLoading] = useState(true)
+	const [fetchCache, setFetchCache] = useState({})
 
 	// Every time dataUrl or dataParser changes, fetch new data and parse it
 	useEffect(() => {
-		(async () => {
+		(() => {
 			setLoading(true)
-			const response = await fetch(dataUrl)
-			const content = await response.text()
-			const parsed = dataParser(content)
-			setData(parsed)
-			setLoading(false)
+
```

Comment:
This chunk is the biggest change in terms of fetching logic, ended up allowing Dataloader to take arrays for `dataUrl`, `dataParser`, and `dataKey` so that it can fetch from multiple endpoints and cache responses as needed.